control-plane/Dockerfile

```diff
@@ -0,0 +1,136 @@
+ARG BIN_NAME=consul-k8s
+ARG VERSION
+
+LABEL name=${BIN_NAME} \
+      maintainer="Team Consul Kubernetes <team-consul-kubernetes@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=${VERSION} \
+      release=${VERSION} \
+      summary="consul-k8s-control-plane provides first-class integrations between Consul and Kubernetes." \
+      description="consul-k8s-control-plane provides first-class integrations between Consul and Kubernetes."
+
+# Set ARGs as ENV so that they can be used in ENTRYPOINT/CMD
+ENV BIN_NAME=${BIN_NAME}
+ENV VERSION=${VERSION}
+
+
+# TARGETOS and TARGETARCH are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+# This is the location of the releases.
+ENV HASHICORP_RELEASES=https://releases.hashicorp.com
+
+# Create a non-root user to run the software.
+RUN addgroup ${BIN_NAME} && \
+    adduser -S -G ${BIN_NAME} 100
+COPY dist/${TARGETOS}/${TARGETARCH}/${BIN_NAME} /bin/
+
+USER 100
+CMD /bin/${BIN_NAME}
+
+# alternate release image, just for the sake of example. In this case we're using
+# debian as the base image just to make the image different from the default alpine one.
+#
+# The use cases for alternate images are things like defining an additional UBI compatible
+# image, or an image with a different function than the main image, e.g. Waypoint's ODR images.
+# -----------------------------------
+# Dockerfile for consul-k8s with UBI as its base image. Used for running on
+# OpenShift.
+#
+# This Dockerfile creates a production release image for the project. This
+# downloads the release from releases.hashicorp.com and therefore requires that
+# the release is published before building the Docker image.
+#
+# We don't rebuild the software because we want the exact checksums and
+# binary signatures to match the software and our builds aren't fully
+# reproducible currently.
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 as release-alternate
+
+# NAME and VERSION are the name of the software in releases.hashicorp.com
+# and the version to download. Example: NAME=consul VERSION=1.2.3.
+ARG BIN_NAME
+ARG VERSION
+
+LABEL name=${BIN_NAME} \
+      maintainer="Team Consul Kubernetes <team-consul-kubernetes@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=${VERSION} \
+      release=${VERSION} \
+      summary="consul-k8s-control-plane provides first-class integrations between Consul and Kubernetes." \
+      description="consul-k8s-control-plane provides first-class integrations between Consul and Kubernetes."
+
+# Set ARGs as ENV so that they can be used in ENTRYPOINT/CMD
+ENV NAME=${BIN_NAME}
+ENV VERSION=${VERSION}
+
+# TARGETOS and TARGETARCH are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+# This is the location of the releases.
+ENV HASHICORP_RELEASES=https://releases.hashicorp.com
+
+# Copy license for Red Hat certification.
+COPY LICENSE.md /licenses/mozilla.txt
+
+RUN microdnf install -y shadow-utils
```

Comment:
Why do we need this? I don't think it was in our current UBI image

Reply:
For some reason 'groupadd' was missing from the ubi image. I pulled it locally and yes it was missing. Very weird.

Reply:
Oh I've missed it in the original image, we do install it there too. I think we also need to install some of those other packages then too:

https://github.com/hashicorp/consul-k8s/blob/d2ce3750c554779139efc2671092f1a8a326db04/control-plane/build-support/docker/Release.ubi.dockerfile#L38

From looking at them, I think some are only needed for installation (like `gnupg`, `wget`, `unzip` and `tar`) so we can probably skip them. But I feel like others we would need baked into the image for sure:
- `iptables` we need this for tproxy
- `ca-certificates` we probably need this to have the latest trusted ca certs 
- `curl` is used by the `tls-init-cleanup` job 
- `libcap` - not sure why we need it but we have it in our release image too
-  `openssl` - not sure what uses it but we have it in our release image too

I think we need to test these images before we merge, wdyt?

Reply:
Also, we probably need to do a similar audit of the utils for the release image

Reply:
That makes sense to me. This means that I will need to get an OpenShift environment running somewhere?

Reply:
For the release and dev images, it should be sufficient to build them, push them to some docker repo and kick off acceptance tests in a draft PR.

For openshift, it's a bit more complicated. I'd still push the openshift image to dockerhub. You can then use [Azure RHO](https://docs.microsoft.com/en-us/azure/openshift/tutorial-create-cluster) to set up a cluster and run tests against it.